src/cdp/CdpConnection.ts

```diff
@@ -162,4 +165,12 @@ export class CdpConnection implements ICdpConnection {
       }
     }
   };
+
+  #createCdpClient(
+    sessionId: Protocol.Target.SessionID | undefined
+  ): CdpClient {
+    const cdpClient = new CdpClient(this, sessionId);
+    this.#sessionCdpClients.set(sessionId, cdpClient);
```

Comment:
Is it possible (however unlikely) this could end up replacing an existing session?

What about `undefined`?

If this is possible, consider checking if it exists before adding it.
But my impression / understanding is that "this should never happen".

Reply:
Undefined session ID points to the main main browser session. Added comment.